spec/fixtures/generate_external_certs.sh

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash
+
+# This script re-generates all of the keys and certs within the external_certs directory.
+# The files there are used by the spec/features/settings/external_cert_feature_spec test
+
+function init() {
+  mkdir -p /tmp/ca_gen/csr
+  CSR_DIR=/tmp/ca_gen/csr
+  
+  mkdir -p /tmp/ca_gen/certs
+  DIR_CERTS=/tmp/ca_gen/certs
+  
+  mkdir -p /tmp/ca_gen/conf
+  DIR_CONF=/tmp/ca_gen/conf
+  
+  mkdir -p /tmp/ca_gen/keys
+  DIR_KEYS=/tmp/ca_gen/keys
+  
+  mkdir -p /tmp/ca_gen/cadata
+  DIR_CADATA=/tmp/ca_gen/cadata
+  
+  echo "dir cadata: $DIR_CADATA"
+}
+
+function cleanup() {
+  rm -rf /tmp/ca_gen
+}
+
+function init_ca() { # $1 ca_name
+  echo "Initiating CA  \"$1\""
+  generate_conf $1
+  # Create / clear certs directory for CA
+  mkdir -p $DIR_CADATA/$1/db.certs
+  rm $DIR_CADATA/$1/db.certs/* 2>/dev/null
+  
+  # Create / clear index file of certs
+  rm $DIR_CADATA/$1/db.inde* 2>/dev/null
+  touch $DIR_CADATA/$1/db.index
+  
+  # Other basic needed files
+  echo unique_subject = yes > $DIR_CADATA/$1/db.index.attr
+  echo 01 > $DIR_CADATA/$1/db.serial
+  
+  # Create the keyfile for the CA
+  openssl genrsa -out $DIR_KEYS/ca_$1.key 2>/dev/null
+}
+
+function init_root_ca() { # $1 ca_name, $2 subject
+  init_ca $1
+  echo "Generating CRT \"$1\""
+  openssl req -new -x509 -days 100 -extensions v3_ca -key $DIR_KEYS/ca_$1.key -out $DIR_CERTS/ca_$1.crt -subj "$2"
+}
+
+function init_sub_ca() { # $1 ca_name, $2 parent_ca, $3 subject
+  init_ca $1
+  echo "Generating CRT \"$1\""
+  openssl req -sha256 -new -key $DIR_KEYS/ca_$1.key -out $CSR_DIR/ca_$1.csr -subj "$3"
+  openssl ca -batch \
+    -config $DIR_CONF/ca_$2.conf \
+    -extensions v3_ca \
+    -out $DIR_CERTS/ca_$1.crt \
+    -infiles $CSR_DIR/ca_$1.csr 2>/dev/null
+}
+
+function append_altnames() { # $1 altnames, $2 filename
+  IFS='-' read -r -a altnames <<< "$1"
```

Comment:
This should have a comment explaining that this just splits a dash delimited string into an array.

Reply:
Added a comment

Reply:
That's a hyphen, not a dash. :p
I actually really like this idiom, though.  And I appreciate the proper use of `@` rather than `*` when the array is used later.  👍